pkg/kn/commands/service/export.go

```diff
@@ -0,0 +1,233 @@
+				return errors.New("'kn service export' requires output format")
+			}
+			serviceName := args[0]
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			client, err := p.NewServingClient(namespace)
+			if err != nil {
+				return err
+			}
+
+			service, err := client.GetService(serviceName)
+			if err != nil {
+				return err
+			}
+
+			history, err := cmd.Flags().GetBool("history")
+			if err != nil {
+				return err
+			}
+
+			// Print out machine readable output if requested
+			if machineReadablePrintFlags.OutputFlagSpecified() {
+				printer, err := machineReadablePrintFlags.ToPrinter()
+				if err != nil {
+					return err
+				}
+				if history {
+					svcList, err := exportServicewithActiveRevisions(service, client)
+					if err != nil {
+						return err
+					}
+					return printer.PrintObj(svcList, cmd.OutOrStdout())
+				}
+				return printer.PrintObj(exportService(service), cmd.OutOrStdout())
+			}
+
+			return nil
+		},
+	}
+	flags := command.Flags()
+	commands.AddNamespaceFlags(flags, false)
+	flags.BoolP("history", "r", false, "Export all active revisions")
+	machineReadablePrintFlags.AddFlags(command)
+	return command
+}
+
+func exportService(latestService *servingv1.Service) *servingv1.Service {
+	return constructServiceTemplate(latestService)
+}
+
+func constructServiceTemplate(latestSvc *servingv1.Service) *servingv1.Service {
+
+	exportedSvc := servingv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   latestSvc.ObjectMeta.Name,
+			Labels: latestSvc.ObjectMeta.Labels,
+		},
+		TypeMeta: latestSvc.TypeMeta,
+	}
+
+	exportedSvc.Spec.Template = servingv1.RevisionTemplateSpec{
+		Spec:       latestSvc.Spec.ConfigurationSpec.Template.Spec,
+		ObjectMeta: latestSvc.Spec.ConfigurationSpec.Template.ObjectMeta,
+	}
+
+	return &exportedSvc
+}
+
+func constructServicefromRevision(latestSvc *servingv1.Service, revision servingv1.Revision) servingv1.Service {
+
+	exportedSvc := servingv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   latestSvc.ObjectMeta.Name,
+			Labels: latestSvc.ObjectMeta.Labels,
+		},
+		TypeMeta: latestSvc.TypeMeta,
+	}
+
+	exportedSvc.Spec.Template = servingv1.RevisionTemplateSpec{
+		Spec:       revision.Spec,
+		ObjectMeta: latestSvc.Spec.ConfigurationSpec.Template.ObjectMeta,
+	}
+
+	exportedSvc.Spec.ConfigurationSpec.Template.ObjectMeta.Name = revision.ObjectMeta.Name
+
+	return exportedSvc
+}
+
+func exportServicewithActiveRevisions(latestSvc *servingv1.Service, client clientservingv1.KnServingClient) (*servingv1.ServiceList, error) {
+	var exportedSvcItems []servingv1.Service
+
+	//get revisions to export from traffic
+	revsMap := getRevisionstoExport(latestSvc)
+
+	var params []clientservingv1.ListConfig
+	params = append(params, clientservingv1.WithService(latestSvc.ObjectMeta.Name))
+
+	// Query for list with filters
+	revisionList, err := client.ListRevisions(params...)
+	if err != nil {
+		return nil, err
+	}
+
+	sortRevisions(revisionList)
```

Comment:
Wondering if an option on sort order would be useful? Perhaps with current as the default?

Reply:
Sort is done to orders the revisions in the order it was created.